modules/transactions/src/main/java/org/apache/ignite/internal/tx/storage/state/rocksdb/TxnStateRocksDbTableStorage.java

```diff
@@ -0,0 +1,343 @@
+    }
+
+    /** {@inheritDoc} */
+    @Override public TxnStateStorage getOrCreateTxnStateStorage(int partitionId) throws StorageException {
+        checkPartitionId(partitionId);
+
+        TxnStateStorage storage = new TxnStateRocksDbStorage(db, writeOptions, readOptions, busyLock, partitionId, this);
+
+        storages.set(partitionId, storage);
+
+        return storage;
+    }
+
+    /** {@inheritDoc} */
+    @Override public @Nullable TxnStateStorage getTxnStateStorage(int partitionId) {
+        return storages.get(partitionId);
+    }
+
+    /** {@inheritDoc} */
+    @Override public CompletableFuture<Void> destroyTxnStateStorage(int partitionId) throws StorageException {
+        checkPartitionId(partitionId);
+
+        TxnStateStorage storage = storages.get(partitionId);
+
+        if (storage != null) {
+            try {
+                storage.close();
+            } catch (Exception e) {
+                throw new StorageException("Couldn't close the transaction state storage of partition "
+                        + partitionId + ", table " + tableCfg.value().name());
+            }
+
+            storage.destroy();
+        }
+
+        return completedFuture(null);
+    }
+
+    /** {@inheritDoc} */
+    @Override public TableConfiguration configuration() {
+        return tableCfg;
+    }
+
+    /** {@inheritDoc} */
+    @Override public void start() throws StorageException {
+        try {
+            flusher = new RocksDbFlusher(
+                busyLock,
+                scheduledExecutor,
+                threadPool,
+                flushDelaySupplier,
+                this::refreshPersistedIndex
+            );
+
+            storages = new AtomicReferenceArray<>(tableCfg.value().partitions());
+
+            this.dbOptions = new DBOptions()
+                    .setCreateIfMissing(true)
+                    .setAtomicFlush(true)
+                    .setListeners(List.of(flusher.listener()));
+
+            this.txDbOptions = new TransactionDBOptions();
+
+            List<ColumnFamilyDescriptor> cfDescriptors;
+
+            try (Options opts = new Options()) {
+                cfDescriptors = RocksDB.listColumnFamilies(opts, dbPath.toAbsolutePath().toString())
+                        .stream()
+                        .map(nameBytes -> new ColumnFamilyDescriptor(nameBytes, new ColumnFamilyOptions()))
+                        .collect(toList());
+
+                cfDescriptors = cfDescriptors.isEmpty()
+                        ? List.of(new ColumnFamilyDescriptor(TXN_STATE_CF.getBytes(StandardCharsets.UTF_8), new ColumnFamilyOptions()))
+                        : cfDescriptors;
+            }
+
+            List<ColumnFamilyHandle> cfHandles = new ArrayList<>(cfDescriptors.size());
+
+            this.db = TransactionDB.open(dbOptions, txDbOptions, dbPath.toString(), cfDescriptors, cfHandles);
+
+            flusher.init(db, cfHandles);
+
+            lastAppliedIndex = readLastAppliedIndex(readOptions);
```

Comment:
Last time, this must belong to a partition, not a table. Please fix it. If you had struggles understanding what it's all about, you should have asked. Current code is absolutely broken.